From 4c13cff4611df72af5aa9219f2f1511cad66f0a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Bonnard?= <frediz@linux.vnet.ibm.com>
Date: Fri, 2 Oct 2015 14:10:16 +0200
Subject: [PATCH] Adding the possibility to disable the proxy

By default, kimchi always starts the nginx proxy itself, and not as
part as the system service. So nginx runs outside of the distro services
and we may not want this when it is supposed to be integrated within
the system : kimchi in distro will be re-using the system's nginx
already running. In Debian/Ubuntu, this is mandatory.
This patch adds the option to start the proxy, and by default it is
disabled.
---
 src/wok.conf.in      |  3 +++
 src/wok/config.py.in |  1 +
 src/wok/proxy.py     | 11 ++++++++++-
 3 files changed, 14 insertions(+), 1 deletion(-)

--- a/src/wok.conf.in
+++ b/src/wok.conf.in
@@ -3,6 +3,9 @@
 #
 
 [server]
+# Start the proxy service?
+#run_proxy = off
+
 # Hostname or IP address to listen on
 #host = 0.0.0.0
 
--- a/src/wok/config.py.in
+++ b/src/wok/config.py.in
@@ -260,6 +260,7 @@
 def _get_config():
     config = SafeConfigParser()
     config.add_section("server")
+    config.set("server", "run_proxy", "off")
     config.set("server", "host", "0.0.0.0")
     config.set("server", "port", "8000")
     config.set("server", "ssl_port", "8001")
--- a/src/wok/proxy.py
+++ b/src/wok/proxy.py
@@ -31,7 +31,7 @@
 
 from wok import sslcert
 from wok.config import paths
-
+import wok.config as config
 
 HTTP_CONFIG = """
 server {
@@ -80,6 +80,9 @@
             with open(key, "w") as f:
                 f.write(ssl_gen.key_pem())
 
+    if config.config.get("server", "run_proxy") == 'off':
+        return
+
     # Setting up Diffie-Hellman group with 2048-bit file
     dhparams_pem = os.path.join(config_dir, "dhparams.pem")
 
@@ -114,6 +117,9 @@
 def start_proxy(options):
     """Start nginx reverse proxy."""
     _create_proxy_config(options)
+    if config.config.get("server", "run_proxy") == 'off':
+        return
+
     nginx_config_dir = paths.nginx_conf_dir
     config_file = "%s/wok.conf" % nginx_config_dir
     cmd = ['nginx', '-c', config_file]
@@ -122,6 +128,9 @@
 
 def terminate_proxy():
     """Stop nginx process."""
+    if config.config.get("server", "run_proxy") == 'off':
+        return
+
     config_file = "%s/wok.conf" % paths.nginx_conf_dir
     term_proxy_cmd = ['nginx', '-s', 'stop', '-c', config_file]
     subprocess.call(term_proxy_cmd)
